pkg/services/otel.go

```diff
@@ -0,0 +1,199 @@
+package services
+
+import (
+	"context"
+	"errors"
+
+	"github.com/obot-platform/obot/pkg/version"
+	"go.opentelemetry.io/contrib/samplers/probability/consistent"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/exporters/otlp/otlplog/otlploggrpc"
+	"go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetricgrpc"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc"
+	"go.opentelemetry.io/otel/log/global"
+	"go.opentelemetry.io/otel/propagation"
+	"go.opentelemetry.io/otel/sdk/log"
+	"go.opentelemetry.io/otel/sdk/metric"
+	"go.opentelemetry.io/otel/sdk/metric/metricdata"
+	"go.opentelemetry.io/otel/sdk/resource"
+	"go.opentelemetry.io/otel/sdk/trace"
+)
+
+type OtelOptions struct {
+	SampleProb         float64 `usage:"The probability of sampling a trace" default:"0.1" name:"otel-sample-prob"`
+	BaseExportEndpoint string  `usage:"The base endpoint to export to, if not set, no metrics, tracing, or logging will be exported" name:"otel-base-export-endpoint"`
+	BearerToken        string  `usage:"Bearer token for authentication" name:"otel-bearer-token"`
```

Comment:
are those option supposed to be set in helm charts too? saw the same thing you did for audit log